doctr/local.py

```diff
@@ -172,3 +173,18 @@ def generate_ssh_key(note, keypath='github_deploy_key'):
 
     with open(keypath + ".pub") as f:
         return f.read()
+
+def check_repo_exists(deploy_repo):
+    """Checks that the deploy repository exists on GitHub before allowing
+    user to generate a key to deploy to that repo.
+    """
+    user, repo = deploy_repo.split('/')
+    search = 'https://api.github.com/search/repositories?q={repo}+user:{user}'
+    r = requests.get(search.format(user=user, repo=repo))
+
+    if r.status_code == requests.codes.unprocessable_entity:
+        sys.exit('User/org "{user}" not found on GitHub.  Exiting'.format(user=user))
+    elif not r.json()['items']:
+        sys.exit('No repo named "{repo}" found for user/org "{user}"'.format(repo=repo, user=user))
```

Comment:
Would rather raise RuntimeError and convert that to a system exit in the command line code.